### ISSUE TITLE
Design: `onyo set --yaml`

### ISSUE THREAD

**Issue by aqw** [psyinfra/onyo#720]
This is the `onyo set` equivalent of #703 

This is part of the general goal to make onyo more consistent across commands (see also #717).

I'm not thrilled about the flag name of `--yaml`, but others aren't any better.

This flag will allow for any yaml document to be passed into it. A template, another asset, or a doc with keys.

Combined with key globbing, this could also allow for some interesting hacks, such as overwriting an asset with a template and re-applying all asset content on top of it (aka: after 5 years, everything is messy, and I want everything to be formatted according to these shiny templates I have).

**Comment by TobiasKadelka** [comment]
The general Idea makes sense, so just +1 on that.

But now I wonder, considering this issue together with #703 in which you say:

> However, despite generating a YAML file, there is no way to pass a YAML file into onyo new.
> Except one can, by abusing --template. A template does not need to be in .onyo/templates. One could pass a YAML file to --template and proceed from there.

Reading both issues together makes me think there should be literally just one flag/option for all of this in `onyo new` and `onyo set`. Because while "template" is very literal for the `onyo new` usecase, it makes otherwise not much sense to add a flag `--yaml` first to `onyo new`, and then a flag `--yaml` to `onyo set` to mirror `onyo new`, and then finally still having a template flag, except maybe as an alias.

Re your "I'm not thrilled about the flag name of `--yaml`" , I agree. It is a bad name, especially since we consider allowing e.g. json files, too. Maybe something more simple/general like `--read`, `--input` or even `--read-file` might do a better job?

**Comment by aqw** [comment]
> Reading both issues together makes me think there should be literally just one flag/option for all of this in `onyo new` and `onyo set`. 

I agree. 

> Maybe something more simple/general like `--read`, `--input` or even `--read-file` might do a better job?

Yeah... I thought of similar things and don't like those either. :laughing: 

I appreciate that `--yaml` is explicit that we pass in YAML. `--read` could be easily a boolean, etc.

**Comment by TobiasKadelka** [comment]
Yeah I mean, in the end `--yaml` is good enough, and if we have jsons, we might want to read them differently anyways as `--json`. 

Either way, I agree on the idea of the issue :)
